File: backend/src/main/java/com/saludtech/dto/UserResponse.java
package com.saludtech.dto;

import lombok.Data;
import java.time.Instant;
import java.util.UUID;

@Data
public class UserResponse {
    private UUID id;
    private String firstName;
    private String lastName;
    private String email;
    private String phone;
    private String documentId;
    private short level;
    private Instant createdAt;
}

File: mobile-app/lib/models/user_response.dart
class UserResponse {
  final String id;
  final String email;
  final String firstName;
  final String lastName;
  final int level;
  final int points;

  UserResponse({
    required this.id,
    required this.email,
    required this.firstName,
    required this.lastName,
    required this.level,
    this.points = 0,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) {
    return UserResponse(
      id: json['id'],
      email: json['email'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      level: json['level'] ?? 1,
      points: json['points'] ?? 0,
    );
  }
}
